### DISCUSSION TITLE
Why is the registration registry decorator in mmengine designed to register_module() manually each time?

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmengine#554] [Q&A]
mmengine 中的注册registry 装饰器为什么要设计成每次手动  register_module() 的形式，为什么不在registry类中添加__call__()函数来解决每次手动注册的问题

Why is the registry decorator in mmengine designed to register_module() manually each time, and why not add __call__() function to the registry class to solve the problem of manual registration each time

![image](https://user-images.githubusercontent.com/25839884/192570376-a2a8ceaa-8933-4d70-9613-cd85ac58371b.png)

**Comment by zhouzaida** [comment]
你好，Registry 实现 `__call__` 方法并调用 `register_module` 同样也可以实现注册功能，但直接调用 `register_module` 会更直观一点。下面是两种用法的对比。

```python
MODEL = Registry('model')

@MODEL()
class ToyModel(nn.Module):
    pass

@MODEL.register_module()
class ToyModel(nn.Module):
    pass
```
